sagemaker/pipeline/pipeline_parameters.py

```diff
@@ -0,0 +1,764 @@
+        # the correct partition input JSON filename?
+        # if "gconstruct" in self.task_config.jobs_to_run:
+        #     if "dist_part" in self.task_config.jobs_to_run or "gb_convert" in self.task_config.jobs_to_run:
+        #         if self.partition_config.input_json_filename != f"{self.task_config.graph_name}.json":
+        #             logging.warning(
+        #                 "When running GConstruct, the partition input JSON "
+        #                 "filename should be '<graph_name>.json'. "
+        #                 "Got %s, setting to %s instead",
+        #                 self.partition_config.input_json_filename,
+        #                 f"{self.task_config.graph_name}.json",
+        #             )
+        #             self.partition_config.input_json_filename = f"{self.task_config.graph_name}.json"
+
+        # Ensure we have a GSProcessing image to run GSProcessing
+        if "gsprocessing" in self.task_config.jobs_to_run:
+            assert (
+                self.aws_config.gsprocessing_pyspark_image_url
+            ), "Need to provide a GSProcessing PySpark image URL when running GSProcessing"
+
+        # Ensure we run gb_convert after dist_part when training with graphbolt
+        if (
+            self.training_config.use_graphbolt_str == "true"
+            and "dist_part" in self.task_config.jobs_to_run
+        ):
+            assert "gb_convert" in self.task_config.jobs_to_run, (
+                "When using Graphbolt and running distributed partitioning, "
+                "need to run 'gb_convert' as a follow-up to 'dist_part', "
+                f"got job sequence: {self.task_config.jobs_to_run}"
+            )
+
+        # When running gconstruct and train with graphbolt enabled, add '--use-graphbolt true'
+        # to gconstruct args
+        if (
+            "gconstruct" in self.task_config.jobs_to_run
+            and self.training_config.use_graphbolt_str == "true"
+        ):
+            self.graph_construction_config.graph_construction_args += (
+                " --use-graphbolt true"
+            )
+
+        # If running gsprocessing but do not set instance count for it, use train instance count
+        if (
+            "gsprocessing" in self.task_config.jobs_to_run
+            and not self.instance_config.gsprocessing_instance_count
+        ):
+            self.instance_config.gsprocessing_instance_count = (
+                self.instance_config.train_infer_instance_count
+            )
+            logging.warning(
+                "No GSProcessing instance count specified, using the training instance count: %s",
+                self.instance_config.gsprocessing_instance_count,
+            )
+
+        # GConstruct uses 'metis', so just translate that if needed
+        if (
+            self.partition_config.partition_algorithm.lower() == "parmetis"
+            and "gconstruct" in self.task_config.jobs_to_run
```

Comment:
first check "gconstruct" in self.task_config.jobs_to_run and then the partition_algorithm.